doc/dev/release.md

```diff
@@ -326,7 +339,7 @@ The release process for the samples repo is simple:
     $ git push --tags
     ```
 
-### (Post-release) Updating the release notes
+### 2. Updating the release notes
 
 Add the following line to the top of the GitHub release notes for `v1.3.0`:
```

Comment:
Now that we're adding a line to the release notes in two places, we probably need to consider how we want those lines formatted after the second line is added.

As is, these instructions would result in the, which is inconsistent and repetitive:

```md
**Note:** See how to upgrade your project to the version v1.3.0 by checking the [Version Upgrade Guide](../migration/version-upgrade-guide.md#v1.3.0)
**NOTE:** ensure the `v1.3.0` tag is referenced when referring to sample code in the [SDK Operator samples repo](https://github.com/operator-framework/operator-sdk-samples/tree/v1.3.0) for this release. Links in SDK documentation are currently set to the samples repo `master` branch.
```

I looked at past releases' notes and we're pretty inconsistent about the formatting and location of the line about the Version Upgrade Guide. And I didn't see any lines about using the correct sample code tag. I think we only started tagging the sample repo somewhat recently, so that might explain that.

My point is, we don't have any working examples for how this _could_ or _should_ look when all is said and done. So I'm open to suggestions.

/cc @estroz

Reply:
IHMO:
- A user will check the info the specific release so will not be respective. 
- Regards he consistence: Could we not just:`**Note:** Ensure the ` for the samples one which will be less effort to change?

Reply:
Hi @joelanford and @estroz, 

The goal of this PR was just to add the steps that we agreed in the ENG meeting 
- Create the JIRA
- + nit regards the NOTEs in the releases 

Could we move forward with as it is and do further updates in others PR if required? OR what you think we should address here to get this merged?

Reply:
Hi @folks,

Should we move forward with it or not? Also, I think we did not create the tasks to upgrade the samples in the latest released which was the goal/problem that we are trying to address here. WDYT?

Reply:
Adding migration guide entries is handled by the changelog/upgrade guide checker now. Instead of all these changes, just make a note about this and LGTM.